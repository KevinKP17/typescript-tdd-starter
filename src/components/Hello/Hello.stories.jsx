import React from 'react';
import Hello from './index'
import { rest } from 'msw'

export default {
    component: Hello,
    title: 'Michael',
};

const Template = args => < Hello {...args }
/>;

export const withPremier = Template.bind({})
withPremier.args = {
    name: "Khea",
    isPremier: true
}

// export const handlers = [
//     rest.get()
// ]