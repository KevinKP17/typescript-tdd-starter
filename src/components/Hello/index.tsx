import React from "react";

export default function Hello({name, isPremier}: HelloType) {
  return (
    <React.Fragment>
      <h1 id="title">{name}</h1>
      { isPremier && <span data-testid="premier-badge">Premier</span> }
    </React.Fragment>
  );
}

type HelloType = {
  name?: string,
  isPremier?: boolean
}

