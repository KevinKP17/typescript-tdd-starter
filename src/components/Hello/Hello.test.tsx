import Hello from "./index";

import React from "react";
import { render, screen } from "@testing-library/react";

test("Component contains premier badge", () => {
  const {asFragment} = render(<Hello name={"Khea"} isPremier={true} />);

  expect(screen.getByRole("heading")).toBeVisible()
  expect(screen.getByTestId("premier-badge")).toBeVisible()

  expect(asFragment()).toMatchSnapshot();
});

test("Component not contain premier badge", () => {
  const {asFragment} = render(<Hello name={"Khea"} isPremier={false} />);

  expect(screen.getByRole("heading")).toBeVisible()
  expect(screen.queryByTestId("premier-badge")).not.toBeInTheDocument()

  expect(asFragment()).toMatchSnapshot();
});
