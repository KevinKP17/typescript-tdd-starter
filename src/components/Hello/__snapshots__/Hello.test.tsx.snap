// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component contains premier badge 1`] = `
<DocumentFragment>
  <h1
    id="title"
  >
    Khea
  </h1>
  <span
    data-testid="premier-badge"
  >
    Premier
  </span>
</DocumentFragment>
`;

exports[`Component not contain premier badge 1`] = `
<DocumentFragment>
  <h1
    id="title"
  >
    Khea
  </h1>
</DocumentFragment>
`;
